File: AGENDLY_APP/lib/features/services/infrastructure/mappers/service_mapper.dart
import 'package:agendly_app/features/services/domain/entities/service.dart';

class ServiceMapper {
  static Service fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['_id'] ?? json['id'],
      businessId: json['businessId'] ?? '',
      name: json['name'] ?? '',
      description: json['description'],
      duration: json['duration'] ?? 60,
      price: (json['price'] ?? 0).toDouble(),
      color: json['color'],
      isActive: json['isActive'] ?? true,
      createdAt: DateTime.parse(json['createdAt'] ?? DateTime.now().toIso8601String()),
      updatedAt: DateTime.parse(json['updatedAt'] ?? DateTime.now().toIso8601String()),
    );
  }

  static Map<String, dynamic> toJson(Service service) {
    return {
      'businessId': service.businessId,
      'name': service.name,
      'description': service.description,
      'duration': service.duration,
      'price': service.price,
      'color': service.color,
      'isActive': service.isActive,
    };
  }
}

File: BACK/src/infrastructure/mappers/service.mapper.ts
import { ServiceEntity } from '../../domain/entities/service.entity';
import { IService } from '../datasources/mongo/models/service.model';

export class ServiceMapper {
  static fromMongo(document: IService): ServiceEntity {
    return {
      id: document._id.toString(),
      businessId: document.businessId,
      name: document.name,
      description: document.description,
      duration: document.duration,
      price: document.price,
      costOfService: document.costOfService,
      category: document.category,
      isActive: document.isActive,
      createdAt: document.createdAt,
      updatedAt: document.updatedAt,
    };
  }
}
